app/models/goal.py

```diff
@@ -3,3 +3,16 @@
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+    def to_dict(self):
+        goal_to_dict = {}
+        goal_to_dict["id"] = self.goal_id
+        goal_to_dict["title"] = self.title
+        return goal_to_dict
+
+    @classmethod
+    def from_dict(cls, goal_data):
+        new_goal = Goal(title=goal_data["title"])
```

Comment:
This works! But what if we changed the name of the model? Now this would continue creating `Goal` which no longer exists, so let's take advantage of the `cls` parameter. This will represent the current model, even if we change its name.

```python
        new_goal = cls(title=goal_data["title"])
```